### PR TITLE
Update deprecated calls, improve WhoDidItBehavior::beforeFind

### PR DESCRIPTION
**Pull request by bancer**
Replace calls to deprecated `config` by `getConfig`.
Allow `WhoDidItBehavior::beforeFind` to construct associations when behavior's 'contain' is `false` but 'CreatedBy' or 'ModifiedBy' are set manually in a specific query's 'contain'.

### CODE REVIEW

**Comment by bobmulder** [comment]
Hi @bancer!

First of all, thank you for your PR. I have to admit I don't work with CakePHP anymore, and barely maintain these packages.

However, if you still use this package I can imagine that this update is prefered.

Please solve the failing tests (see https://travis-ci.org/cakemanager/cakephp-utils/jobs/581185857), so I can merge this PR.

Thanks! Bob

**Comment by bancer** [comment]
Hi @bobmulder,
I updated all deprecated calls. All tests pass now.

**Comment by bobmulder** [comment]
Great work! Thanks for your fixes!

**Comment by bancer** [comment]
@bobmulder I think other 3 pull requests can be closed now.

**Comment by martonmiklos** [comment]
Many thanks @bancer  for integrating the fixes and @bobmulder  for the merge! I have closed my PR which implemented a subset of  these changes.

**Comment by bobmulder** [comment]
@bancer & @martonmiklos; if any of you is interested in becoming a contributor please let me know! Like I said before I don't work with CakePHP anymore, and barely maintain these packages.

**Comment by bancer** [comment]
Yes, sure. Why not?

**Comment by martonmiklos** [comment]
tldr: count me in!

I do not work with CakePHP on daily basis, but I use a couple of your plugins from this repo what I will need to keep deprecation free for a while and I am willing to put effort to this.

**Comment by bobmulder** [comment]
Always nice to hear my work is used somewhere :) 

At the same time this is the pitfall of open source. They're quickely not maintained without a community on it. 

@martonmiklos I'll add you to the repo!
